src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -499,9 +514,19 @@ public void Set(ReadOnlySpan<byte> key, byte[]? value, WriteFlags flags = WriteF
 
     public WriteOptions? WriteFlagsToWriteOptions(WriteFlags flags)
     {
-        if (flags == WriteFlags.LowPriority)
+        if ((flags & WriteFlags.LowPriority) != 0 && (flags & WriteFlags.DisableWAL) != 0)
```

Comment:
Either save flags to `bool` and reuse them or check for both flags in one go.

Reply:
Ok